include/deal.II/matrix_free/cuda_matrix_free.h

```diff
@@ -151,7 +156,35 @@ namespace CUDAWrappers
      * degrees of freedom, the DoFHandler and the mapping describe the
      * transformation from unit to real cell, and the finite element
      * underlying the DoFHandler together with the quadrature formula
-     * describe the local operations.
+     * describe the local operations. This function supports distributed
+     * computation (MPI).
+     */
+    void
+    reinit(const Mapping<dim> &             mapping,
+           const DoFHandler<dim> &          dof_handler,
+           const AffineConstraints<Number> &constraints,
+           const Quadrature<1> &            quad,
+           std::shared_ptr<const MPI_Comm>  comm,
```

Comment:
why `comm` as a shared pointer?

Reply:
Because I don't want to create a partitioner if I don't have an mpi communicator.

Reply:
so it kind of combines serial and MPI-parallel cases? 

I just don't recall such approach being used elsewhere in `deal.II`, but I am fine with it.

Reply:
Yeah, basically. If you prefer I can remove it from the interface and have a private function with this interface.

Reply:
> If you prefer I can remove it from the interface and have a private function with this interface.

yes, I think that would be more consistent with the rest of the library.